### PR TITLE
add cloudinary package

### PR DESCRIPTION
**Pull request by jbevis**
This PR adds the ```next-tinacms-cloudinary``` package as well as testing a use case within the cloud starter demo.

### CODE REVIEW

File: examples/tina-cloud-starter/components/tina-wrapper.tsx [hunk 1]
```diff
@@ -76,4 +73,82 @@ const Inner = (props) => {
...
+
+  constructor(public cloud_name: string) {
+    this.api = new Cloudinary({
+      cloud_name: this.cloud_name,
+      secure: true,
+    })
+  }
+
+  async persist(media: MediaUploadOptions[]): Promise<Media[]> {
+    const url = `https://api.cloudinary.com/v1_1/${process.env.NEXT_PUBLIC_CLOUDINARY_CLOUD_NAME}/image/upload`
```
Comment (jeffsee55) [inline examples/tina-cloud-starter/components/tina-wrapper.tsx:97]: Should this be:
```suggestion
    const url = `https://api.cloudinary.com/v1_1/${this.cloud_name}/image/upload`
```

File: examples/tina-cloud-starter/components/tina-wrapper.tsx [hunk 2]
```diff
@@ -76,4 +73,82 @@ const Inner = (props) => {
...
+  MediaUploadOptions,
+} from 'tinacms'
+
+export class CloudinaryMediaStore2 implements MediaStore {
+  accept = '*'
+  private api: Cloudinary
+
+  constructor(public cloud_name: string) {
+    this.api = new Cloudinary({
+      cloud_name: this.cloud_name,
```
Comment (jeffsee55) [inline examples/tina-cloud-starter/components/tina-wrapper.tsx:91]: Do we need to set `this.cloud_name = cloud_name` first?

File: examples/tina-cloud-starter/components/tina-wrapper.tsx [hunk 3]
```diff
@@ -87,4 +84,82 @@ const Inner = (props) => {
...
+import { Cloudinary } from 'cloudinary-core'
+import {
+  Media,
+  MediaList,
+  MediaListOptions,
+  MediaStore,
+  MediaUploadOptions,
+} from 'tinacms'
+
+export class CloudinaryMediaStore2 implements MediaStore {
```
Comment (jeffsee55) [inline examples/tina-cloud-starter/components/tina-wrapper.tsx:96]: Is this code needed?

File: examples/tina-cloud-starter/components/tina-wrapper.tsx (hunk 1 above)
Comment (jbevis) [reply examples/tina-cloud-starter/components/tina-wrapper.tsx:108]: So, actually this declaration isn't actually getting used anywhere. Must have missed pruning it, and I think that was the only instance of using a hard coded URL.

File: examples/tina-cloud-starter/components/tina-wrapper.tsx (hunk 2 above)
Comment (jbevis) [reply examples/tina-cloud-starter/components/tina-wrapper.tsx:102]: Oh.... that's not supposed to be there. Some old debugging code.

File: examples/tina-cloud-starter/components/tina-wrapper.tsx (hunk 3 above)
Comment (jbevis) [reply examples/tina-cloud-starter/components/tina-wrapper.tsx]: It is not, was some debugging code I forgot to remove. Just pushed up a removal.

File: examples/tina-cloud-starter/components/tina-wrapper.tsx (hunk 2 above)
Comment (jeffsee55) [reply examples/tina-cloud-starter/components/tina-wrapper.tsx]: Yeah sorry I realize that this code was elshewhere after I commented

File: packages/next-tinacms-cloudinary/src/handlers/create-media-handler.ts [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+export const createMediaHandler = (config: CloudinaryConfig) => {
+  cloudinary.config(config)
+
+  return async (req: NextApiRequest, res: NextApiResponse) => {
+    switch (req.method) {
+      case 'GET':
+        return listMedia(req, res)
+      case 'POST':
+        return uploadMedia(req, res)
```
Comment (jeffsee55) [inline packages/next-tinacms-cloudinary/src/handlers/create-media-handler.ts:28]: I think this is all we need to totally remove the `/media/[public_id].ts` file:
```suggestion
        return uploadMedia(req, res)
    case 'DELETE':
      const { media } = req.query
      const [_media, public_id] = media
      cloudinary.uploader.destroy(public_id as string, {}, (err) => {
        if (err) res.status(500)
        res.json({
          err,
          public_id,
        })
      })
```

File: packages/next-tinacms-cloudinary/src/handlers/create-media-handler.ts [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+
+  cloudinary.uploader.destroy(public_id as string, {}, (err) => {
+    if (err) res.status(500)
+    res.json({
+      err,
+      public_id,
+    })
+  })
+}
+
```
Comment (jbevis) [inline packages/next-tinacms-cloudinary/src/handlers/create-media-handler.ts:124]: @jeffsee55 that did it. 

Added it as a `deleteAsset` function to create-media-hander.ts

File: packages/next-tinacms-cloudinary/src/handlers/create-media-handler.ts (hunk 4 above)
Comment (jbevis) [reply packages/next-tinacms-cloudinary/src/handlers/create-media-handler.ts:28]: Latest changes should update [...media].ts as a catch all (GET, POST, DELETE)

Also, cleans up the media store class and removes the useless constructor.

**Comment by tpiros** [comment]
> /cc @tpiros in case you or someone from Cloudinary want to take a look 👀

I'm going to take a look at some point next week :) Thanks for the mention @DirtyF

File: packages/next-tinacms-cloudinary/README.md [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+Now, you can register the Cloudinary Media store with the instance of Tina in your app.
+
+```
+import { CloudinaryMediaStore } from 'next-tinacms-cloudinary'
+
+...
+const cms = new TinaCMS({
+  ---
+  media: new CloudinaryMediaStore(
+    process.env.NEXT_PUBLIC_CLOUDINARY_CLOUD_NAME
```
Comment (jeffsee55) [inline packages/next-tinacms-cloudinary/README.md:33]: Looks like this is no longer necessary

**Review by jeffsee55 — APPROVED** [review]
Looks great, one small comment on docs

**Pull request merged**